core-primitives/storage/src/storage_entry.rs

```diff
@@ -1,49 +1,10 @@
 use crate::{error::Error, StorageProofChecker};
-use codec::{Decode, Encode};
+use codec::Decode;
 use frame_support::ensure;
+use itp_types::storage::{StorageEntry, StorageEntryVerified};
 use sp_runtime::traits::Header as HeaderT;
 use sp_std::prelude::Vec;
 
-#[derive(Default, Clone, Encode, Decode)]
-pub struct StorageEntry<V> {
-	pub key: Vec<u8>,
-	pub value: Option<V>,
-	pub proof: Option<Vec<Vec<u8>>>,
-}
```

Comment:
Now that `StorageEntry` is moved out of this file, we should probably re-name the file to better reflect what is in it? E.g. `verify_storage_proof.rs`